doc/manual/src/language/builtin-constants.md

```diff
@@ -1,20 +1,42 @@
 # Built-in Constants
 
-Here are the constants built into the Nix expression evaluator:
-
-  - `builtins`\
-    The set `builtins` contains all the built-in functions and values.
-    You can use `builtins` to test for the availability of features in
-    the Nix installation, e.g.,
-    
-    ```nix
-    if builtins ? getEnv then builtins.getEnv "PATH" else ""
-    ```
-    
-    This allows a Nix expression to fall back gracefully on older Nix
-    installations that don’t have the desired built-in function.
-
-  - [`builtins.currentSystem`]{#builtins-currentSystem}\
-    The built-in value `currentSystem` evaluates to the Nix platform
-    identifier for the Nix installation on which the expression is being
-    evaluated, such as `"i686-linux"` or `"x86_64-darwin"`.
+These constants are built into the Nix language evaluator:
+
+- [`builtins`]{#builtins-builtins} (attribute set)
+
+  Contains all the [built-in functions](./builtins.md) and values, in order to avoid polluting the global scope.
+
+  Since built-in functions were added over time, [testing for attributes](./operators.md#has-attribute) in `builtins` can be used for graceful fallback on older Nix installations:
+
+  ```nix
+  if builtins ? getEnv then builtins.getEnv "PATH" else ""
+  ```
+
+- [`builtins.currentSystem`]{#builtins-currentSystem} (string)
+
+  The built-in value `currentSystem` evaluates to the Nix platform
+  identifier for the Nix installation on which the expression is being
+  evaluated, such as `"i686-linux"` or `"x86_64-darwin"`.
+
+- [`builtins.currentTime`]{#builtins-currentTime} (integer)
+
+  Return the [Unix time](https://en.wikipedia.org/wiki/Unix_time) at first evaluation.
+  Repeated references to that name will re-use the initially obtained value.
+
+  Example:
+
+  ```console
+  $ nix repl
+  Welcome to Nix 2.15.1 Type :? for help.
+
+  nix-repl> builtins.currentTime
+  1683705525
+
+  nix-repl> builtins.currentTime
+  1683705525
+  ```
+
+  > **Warning**
+  >
+  > This is intended for testing and debugging only.
+  > Do not use it in production systems, since an impure, that is, non-constant value will break reproducibility.
```

Comment:
Should mention that neither of these is available in pure eval mode (i.e. the default for `nix`).

The main use case for `currentTime` is non-deterministic derivations. These days, that is better done using the `impure-derivations` experimental feature.